tools/please_pex/pex/pex_main.py

```diff
@@ -250,10 +250,9 @@ def _has_distribution(self):
                         if name and self._match_file(name, ""):
                             return True
 
-        if context.name in sys.modules:
```

Comment:
There'll be a performance impact associated with removing this guard because `PexDistribution` will now go searching through the zip file every time distribution metadata is requested, although I'm not sure whether we care about the speed of those lookups. We could pay this price once at startup and cache the names of the distributions stored in the pex file (or perhaps even build that list at pex build-time, if we care sufficiently).